syntax = "proto3";

package pslx;

message EmptyMessage {
}

// the next will be 4
message Credentials {
    string user_name = 1;
    string password = 2;
    map<string, string> others = 3;
}

// the next will be 4
message FileInfo {
    string file_path = 1;
    string file_size = 2;
    string modified_time = 3;
}

// the next will be 9
message FrontendConfig {
    message ContainerBackendConfig {
        string server_url = 1;
        string root_certificate_path = 2;
    }
    message ViewerConfig {
        map<string, string> server_url_to_root_certificate_map = 3;
    }
    string sqlalchemy_database_uri = 4;
    ContainerBackendConfig container_backend_config = 5;
    ViewerConfig proto_viewer_config = 6;
    ViewerConfig file_viewer_config = 7;

    Credentials credential = 8;
}
